### DISCUSSION TITLE
looking for some clarity on a few small issues

### DISCUSSION THREAD

**Discussion by isaacgolding** [Security-Onion-Solutions/securityonion#7844] [Unsupported Versions] [answered]
So Ive been in a "test lab" for about two weeks now and my current setup is as follows:
1) Manager+Search           2.) Forward
**My first issue** is where should I be pointing my agents?   They are all currently pointed at the manager.

**My 2nd issue** is when I attempt to use so-allow on the forward device I get the following response:

> Adding 10.23.83.0/24 to the beats_endpoint role. This can take a few seconds...
> Traceback (most recent call last):
>   File "/sbin/so-firewall", line 392, in <module>
>     main()
>   File "/sbin/so-firewall", line 382, in main
>     code = cmd(options, args[1:])
>   File "/sbin/so-firewall", line 303, in includehost
>     result = addIp(args[0], args[1], 'insert')
>   File "/sbin/so-firewall", line 101, in addIp
>     content = loadYaml(hostgroupsFilename)
>   File "/sbin/so-firewall", line 73, in loadYaml
>     file = open(filename, "r")
> FileNotFoundError: [Errno 2] No such file or directory: '/opt/so/saltstack/local                                                                                             /salt/firewall/hostgroups.local.yaml'

**My 3rd issue**  What is the proper way to test Strelka?   The so-status indicates its running and I'm not seeing any indicators that it is not running other than the lack of any output into Hunt or Kibana screens on the web interface.

**My 4th (and last) issue**  I see that software inventory and vuln testing is disabled. And from reading elsewhere I understand that it has some issue with the vuln testing bits.    But at this point since I am running this system in a testing phase I'd rather have the software inv/vuln testing enabled if no other reason than to have it running and generating it's data.   Is this something that I can fix by editing a config file or 3 or is there more to it than that?

**Comment by dougburks** [comment]
> 1. **My first issue** is where should I be pointing my agents?   They are all currently pointed at the manager.

If that works for you, then that should be fine. Depending on what agent you're using, how many agents you're deploying, and what your network topology looks like, you have some flexibility to point agents to a different node if necessary.


> **My 2nd issue** is when I attempt to use so-allow on the forward device I get the following response:

From https://docs.securityonion.net/en/2.3/firewall.html#managing:
_Managing firewall rules, for all devices, should be done from the manager node using either [so-allow](https://docs.securityonion.net/en/2.3/so-allow.html#so-allow), so-firewall or, for advanced cases, manually editing the yaml files._

> **My 3rd issue** What is the proper way to test Strelka? The so-status indicates its running and I'm not seeing any indicators that it is not running other than the lack of any output into Hunt or Kibana screens on the web interface.

You could manually download a malware sample via HTTP or perhaps import a malware pcap where malware was downloaded via HTTP. The Poison Ivy RAT example at https://docs.securityonion.net/en/2.3/strelka.html is from the pcap sample at https://github.com/markofu/pcaps/tree/master/jackcr.

> **My 4th (and last) issue** I see that software inventory and vuln testing is disabled. And from reading elsewhere I understand that it has some issue with the vuln testing bits. But at this point since I am running this system in a testing phase I'd rather have the software inv/vuln testing enabled if no other reason than to have it running and generating it's data. Is this something that I can fix by editing a config file or 3 or is there more to it than that?

Which agent are you referring to?

**Reply by isaacgolding** [reply]
Sorry for missing the agent.    FleetDM.  I'd like to turn on software inventory if it's not a major issue. And yes I know its not quite mature yet.

**Comment by defensivedepth** [comment, marked as answer]
@isaacgolding What version of SO are you running?


To manually enable the software inventory, you can do the following:

On the Manager:

#### This will give you a prompt inside the Fleet container
sudo docker exec -it so-fleet /bin/ash

#### Set fleetctl config
fleetctl config set --address https://127.0.0.1:8080/ --tls-skip-verify --url-prefix /fleet

####  Login to fleet via fleetctl - the email and password you created during so-setup.
fleetctl login

####  Copy the current config to a file called fleet.config
fleetctl get config > fleet.config

####  Change the Vulnerability setting to True
sed -i 's/enable_software_inventory: false/enable_software_inventory: true/' fleet.config

####  Apply the new config
fleetctl apply -f fleet.config

####  Exit the container & restart Fleet
exit
sudo so-fleet-restart --force

**Reply by isaacgolding** [reply]
I’m using 2.3.120. Thanks for the directions.  I’ve run them and an inventory is happening now.   Of course my follow up is going to be on the CVE setup. But I’m going to attempt to figure that out on my own first.
